nsls2_detector_handlers/electrometer.py

```diff
@@ -0,0 +1,61 @@
+import numpy as np
+import os
+import pandas as pd
+
+from . import HandlerBase
+
+
+class ElectrometerBinFileHandler(HandlerBase):
+    """Read electrometer *.bin files"""
+
+    def __init__(self, fpath):
+        # It's a text config file, which we don't store in the resources yet, parsing for now
+        fpath_txt = f'{os.path.splitext(fpath)[0]}.txt'
+        self.filename = fpath
```

Comment:
Want to draw attention to this:

Do we want `fpath` or `fpath_txt` to be returned by `get_file_list`?

Reply:
we should return both `fpath_txt` and all of the files in it in `get_file_list`.  

One (the only?) use for `get_file_list` is so that we can move these files around behind the scenes without re-writing them to a new format.  To actually capture all of the data we need to know what files it is going to touch.

Reply:
Actually we may need both, but the files with the data are represented by `fpath` (it's a custom `.bin` file produced by the device.)

Reply:
FYI, `fpath_txt` is actually a file which has some metadata needed to properly read the `fpath` (`.bin`) file. I didn't bother to create a separate resource/datum for it, neither added any contents from `.txt` into the mongo documents. It's an associated file which we assume is in place with the same base name and different extension.

Here is an example of such files:
- https://github.com/NSLS-II-ISS/profile_collection/blob/master/startup/scripts/future-handler/FAstreamSettings.txt
- https://github.com/NSLS-II-ISS/profile_collection/blob/master/startup/scripts/future-handler/FAstream.bin (the base names are different here, but they are the same when produced in a scan)
- and the script to read them https://github.com/NSLS-II-ISS/profile_collection/blob/master/startup/scripts/future-handler/binaryRead.py (which was ported to the handler code)

Reply:
> FYI, fpath_txt is actually a file which has some metadata needed to properly read the fpath (.bin) file. I didn't bother to create a separate resource/datum for it, neither added any contents from .txt into the mongo documents. It's an associated file which we assume is in place with the same base name and different extension.

Thank you for the context!

Thinking this over, this sounds like a potential 'gotcha' for our efforts to transfer data to SDCC. If I understand correctly, these files need to be transferred as well, but I don't think any part of the current transfer mechanism would pick them up... unless including it in the `get_file_list` return takes care of that? Maybe @danielballan can confirm?

Reply:
Yes, what we copy over is controlled by what is in `get_file_list` so it is both necessary and sufficient to include the file path there to get it copied over the road.